core/src/main/java/org/mskcc/cbio/portal/scripts/ImportMutationalSignatureMetaData.java

```diff
@@ -0,0 +1,184 @@
+        }
+        catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
+    /**
+     * Adds the mutational signatures parsed from the file into the Database.
+     * This is for importing mutational signature metadata
+     * MetaData file has first column as mutational_signature_id and second column as description, each row is a new signature
+     * @param mutationalSignatureMetaFile File with mutational signature information
+     * @throws IOException
+     * @throws DaoException
+     */
+    
+    public static void importData (File mutationalSignatureMetaFile, boolean replace) throws IOException, DaoException{
+        try{
+            FileReader reader = new FileReader(mutationalSignatureMetaFile);
+            BufferedReader buf = new BufferedReader(reader);
+            String line;
+            while ((line = buf.readLine()) != null){
+                ProgressMonitor.incrementCurValue();
+                ConsoleUtil.showProgress();
+                
+                
+                String[] parts = line.split("\t");
+                boolean arrayIsEmpty = false;
+                for(int i = 0; i < parts.length; i++){
+                    if (parts[i].equals("") || parts[i] == null){
+                        arrayIsEmpty = true;
+                    }
+                    else{
+                        arrayIsEmpty = false;
+                    }
+                }
+
+                if (!arrayIsEmpty){
+                    //assume mutational signature id and description are first two columns in file
+                    MutationalSignatureMeta mutationalSignatureMeta = new MutationalSignatureMeta();
+                    mutationalSignatureMeta.setMutationalSignatureId(parts[0]);
+                    mutationalSignatureMeta.setDescription(parts[1]);
+
+                    //check if mutational signature already exists in database by the mutational signature id
+                    MutationalSignatureMeta existingMutationalSignature = DaoMutationalSignature.getMutationalSignatureById(mutationalSignatureMeta.getMutationalSignatureId());
+                    
+                    if (existingMutationalSignature == null){ //add a new mutational signature if database does not contain it
+                        ProgressMonitor.setCurrentMessage("Adding: " + mutationalSignatureMeta.getMutationalSignatureId());
+                        DaoMutationalSignature.addMutationalSignature(mutationalSignatureMeta);
+                    }
+                    else{//if it already exists, either replace description or do nothing
+                        if (replace){
+                            ProgressMonitor.setCurrentMessage("Updating mutational signature: " + parts[0]);
+                            DaoMutationalSignature.updateMutationalSignature(mutationalSignatureMeta);
+                            }
```

Comment:
I would add a ProgressMonitor statement that says signature exists, but replace flag is false, so skipping.